app/assets/scripts/views/region.js

```diff
@@ -0,0 +1,597 @@
+import BlockLoading from '../components/block-loading';
+
+class Region extends React.Component {
+  constructor (props) {
+    super(props);
+    this.state = {
+      pageAppeals: 1,
+      pageDrefs: 1,
+      pageReports: 1
+    };
+    this.handlePageChange = this.handlePageChange.bind(this);
+  }
+
+  componentWillReceiveProps (nextProps) {
+    if (this.props.match.params.id !== nextProps.match.params.id) {
+      this.getData(nextProps);
+      return this.getRegion(nextProps.match.params.id);
+    }
+
+    if (this.props.region.fetching && !nextProps.region.fetching) {
+      hideGlobalLoading();
+      if (nextProps.region.error) {
+        this.props.history.push('/uhoh');
+      }
+    }
+  }
+
+  componentDidMount () {
+    this.getData(this.props);
+    this.getRegion(this.props.match.params.id);
+  }
+
+  getData (props) {
+    this.props._getRegionAppeals(props.match.params.id);
+    this.props._getRegionDrefs(props.match.params.id);
+    this.props._getRegionFieldReports(props.match.params.id);
+    this.props._getRegionAppealsStats(props.match.params.id);
+    this.props._getRegionAggregateAppeals(props.match.params.id, DateTime.local().minus({years: 10}).startOf('month').toISODate(), 'year');
+    this.props._getRegionERU(props.match.params.id);
+  }
+
+  getRegion (id) {
+    showGlobalLoading();
+    this.props._getRegionById(id);
+  }
+
+  handlePageChange (what, page) {
+    let pageKey;
+    let fn;
+    switch (what) {
+      case 'appeals':
+        pageKey = 'pageAppeals';
+        fn = this.props._getRegionAppeals;
+        break;
+      case 'drefs':
+        pageKey = 'pageDrefs';
+        fn = this.props._getRegionDrefs;
+        break;
+      case 'fieldReports':
+        pageKey = 'pageReports';
+        fn = this.props._getRegionFieldReports;
+        break;
+    }
+    this.setState({ [pageKey]: page.selected + 1 }, () => {
+      fn(this.props.match.params.id, this.state[pageKey]);
+    });
+  }
+
+  renderAppeals () {
+    const {
+      fetched,
+      fetching,
+      error,
+      data
+    } = this.props.appeals;
+
+    if (fetching) {
+      return (
+        <Fold title='Appeals'>
+          <BlockLoading/>
+        </Fold>
+      );
+    }
+
+    if (error) {
+      return (
+        <Fold title='Appeals'>
+          <p>Oh no! An error ocurred getting the data.</p>
+        </Fold>
+      );
+    }
+
+    if (fetched) {
+      const now = Date.now();
+      if (data && data.objects.length) {
+        const headings = [
+          { id: 'date', label: 'Date' },
+          { id: 'name', label: 'Name' },
+          { id: 'event', label: 'Event' },
+          { id: 'dtype', label: 'Disaster Type' },
+          { id: 'requestAmount', label: 'Appeal Amount (CHF)' },
+          { id: 'fundedAmount', label: 'Funding (CHF)' },
+          { id: 'active', label: 'Active' }
+        ];
+
+        const rows = data.objects.map(o => ({
+          id: o.id,
+          date: DateTime.fromISO(o.created_at).toISODate(),
```

Comment:
I think for appeals and drefs, either `start_date` or `end_date` would be a better proxy for this than created_at, which will all be the same right now. If I were to pick one it'd probably be `end_date`.

Reply:
Changed for appeals and drefs. What about field reports?